### PR TITLE
Ports station traits from TGstation

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

Port of https://github.com/tgstation/tgstation/pull/56309

- [x] port the actual code
- [x] just disable hangover until someone maps in the spawn points lol

## Why It's Good For The Game

It's cool.

## Changelog
:cl:
add: Adds station traits: Small modifiers that can randomly be chosen each round. They are usually logged in the intercept the station receives early in the round
refactor: Announcement audio is now stored in a datum to make it easy to add different announcers
/:cl:

### CODE REVIEW

**Comment by Putnam3145** [comment]
why is it like this. if this isn't squash merged i will end you

**Comment by BlueWildrose** [comment]
There's a pretty cool variety here, I like it.

**Comment by LordSaladin** [comment]
We could add some server specific traits like a golden [REDACTED] spawning in maint

**Review by DeltaFire15 — CHANGES_REQUESTED** [review]
I know that hangover is not enabled yet, but to really make it "done but no mapping done yet", please nonetheless disable Synthetics from being affected by it as they do not process alcohol and it'd be impossible for them to suffer from it. 
Should probably be good to testmerge after.

**Comment by DeltaFire15** [comment]
Unaddressed merge conflicts and requested changes. Marking PR as stale.

**Comment by Putnam3145** [comment]
ping me on discord, notifications on github are not very good for me since I get them for every single comment on every single PR opened here

**Comment by Putnam3145** [comment]
> I know that hangover is not enabled yet, but to really make it "done but no mapping done yet", please nonetheless disable Synthetics from being affected by it as they do not process alcohol and it'd be impossible for them to suffer from it. Should probably be good to testmerge after.

also, i hate to say this, but, uh, actually robots do process alcohol. this might be an oversight of some sort but there's nothing preventing them at all, from what I can tell. I'm fixing it anyway, but yeah

File: code/__HELPERS/priority_announce.dm [hunk 1]
```diff
@@ -170,7 +81,7 @@
 		title = "Classified [command_name()] Update"
 
 	if(announce)
-		priority_announce("A report has been downloaded and printed out at all communications consoles.", "Incoming Classified Message",  "commandreport", has_important_message = TRUE)
+		priority_announce("A report has been downloaded and printed out at all communications consoles.", "Incoming Classified Message", "commandreport", SSstation.announcer.get_rand_report_sound(), has_important_message = TRUE)
```
Comment (deathride58) [inline code/__HELPERS/priority_announce.dm]: These args are malformed; command report sounds are muted as a result ("commandreport" is no longer an announcer event sound).

**Review by deathride58 — CHANGES_REQUESTED** [review]
Only one tiny bug before it can be merged proper

**Comment by deathride58** [comment]
Remaining oddities:
- "Modify station traits" has no TGUI at all; it simply returns "Interface StationTraitsPanel was not found."
- Command reports.... Somehow still don't have sound? Very weird.

**Pull request merged**